{
  expectCall(retryable());
  expectCall(retryable());
  Throwable exception=retryable();
  expectCall(exception);
  control.replay();
  RetryingCaller retry=makeRetry(false,RETRYABLE);
  try {
    call(retry);
    fail();
  }
 catch (  Throwable t) {
    assertThat(t,is(exception));
  }
  assertThat(memoizeGetCounter.get(methodA).get(),is(2L));
}
